packages/console/src/monaco/MonacoProviders.tsx

```diff
@@ -0,0 +1,346 @@
+      case 25:
+        return monacoKinds.TypeParameter;
+      default:
+        return monacoKinds.Variable;
+    }
+  }
+
+  constructor(props: MonacoProviderProps) {
+    super(props);
+
+    this.handleCompletionRequest = this.handleCompletionRequest.bind(this);
+    this.handleSignatureRequest = this.handleSignatureRequest.bind(this);
+    this.handleHoverRequest = this.handleHoverRequest.bind(this);
+  }
+
+  componentDidMount(): void {
+    const { language } = this.props;
+
+    this.registeredCompletionProvider = monaco.languages.registerCompletionItemProvider(
+      language ?? '',
+      {
+        provideCompletionItems: this.handleCompletionRequest,
+        triggerCharacters: ['.'],
+      }
+    );
+
+    this.registeredSignatureProvider = monaco.languages.registerSignatureHelpProvider(
+      language ?? '',
+      {
+        provideSignatureHelp: this.handleSignatureRequest,
+        signatureHelpTriggerCharacters: ['(', ','],
+      }
+    );
+
+    this.registeredHoverProvider = monaco.languages.registerHoverProvider(
+      language ?? '',
+      {
+        provideHover: this.handleHoverRequest,
+      }
+    );
+  }
+
+  componentWillUnmount(): void {
+    this.registeredCompletionProvider?.dispose();
+    this.registeredSignatureProvider?.dispose();
+    this.registeredHoverProvider?.dispose();
+  }
+
+  registeredCompletionProvider?: monaco.IDisposable;
+
+  registeredSignatureProvider?: monaco.IDisposable;
+
+  registeredHoverProvider?: monaco.IDisposable;
+
+  handleCompletionRequest(
+    model: monaco.editor.ITextModel,
+    position: monaco.Position,
+    context: monaco.languages.CompletionContext
+  ): monaco.languages.ProviderResult<monaco.languages.CompletionList> {
+    const { model: propModel, session } = this.props;
+    if (model !== propModel) {
+      return null;
+    }
+
+    const params = {
+      textDocument: {
+        uri: `${model.uri}`,
+        version: model.getVersionId(),
+      },
+      // Monaco 1-indexes Position. LSP 0-indexes Position
+      position: {
+        line: position.lineNumber - 1,
+        character: position.column - 1,
+      },
+      context,
+    };
+
+    const completionItems = session.getCompletionItems(params);
+
+    const monacoCompletionItems = completionItems
+      .then(items => {
+        log.debug('Completion items received: ', params, completionItems);
+
+        // Annoying that the LSP protocol returns completion items with a range that's slightly different than what Monaco expects
+        // Need to remap the items here
+        const suggestions = items.map(item => {
+          const {
+            label,
+            kind,
+            detail,
+            documentation,
+            sortText,
+            filterText,
+            textEdit,
+            insertTextFormat,
+          } = item;
+
+          const { start, end } = textEdit.range;
+
+          // Monaco expects the columns/ranges to start at 1.
+          const range = {
```

Comment:
Could probably put this in a `lspToMonacoRange` method, similar to `lspToMonacoKind` to clean it up a little bit.